src/catch2/internal/catch_assertion_handler.cpp

```diff
@@ -22,13 +25,23 @@ namespace Catch {
     :   m_assertionInfo{ macroName, lineInfo, capturedExpression, resultDisposition },
         m_resultCapture( getResultCapture() )
     {
+        auto lock = take_global_lock();
         m_resultCapture.notifyAssertionStarted( m_assertionInfo );
     }
 
+    AssertionHandler::~AssertionHandler() {
+        auto lock = take_global_lock();
```

Comment:
Can this be moved inside the `if` or is `m_completed` written to from other threads?